### DISCUSSION TITLE
issue: `useForm` and `useFormContext` without `FieldValues` gives `formState.errors` object with `any` instead of `FieldError`

### DISCUSSION THREAD

**Discussion by Svish** [react-hook-form/react-hook-form#8518] [Q&A] [answered]
### Version Number

7.32.1

### Codesandbox/Expo snack

https://codesandbox.io/s/error-type-issue-with-react-hook-form-ku968b

### Steps to reproduce

1. Use Typescript
2. Write a generic component to list all errors in a form
3. Use `useFormContext`, but do not pass a type for the values, since this component is generic and we don't know which fields are there.
4. Try to loop over the values in the `errors` object from `formState`

Notice that you're getting warnings about unsafe `any` because the error type for some reason ends up as `any` instead of as `FieldError`.

### Expected behaviour

Expected the type to be the same as it normally is, except for keys to be unknown:

```ts
const errors: {
    [x: string]: FieldError | undefined;
}
```

But instead it ends up as the following, which gives warnings from Typescript and no type-help or type-safety from editor:

```ts
const errors: {
    [x: string]: any;
}
```

If I pass a type for the values, then it becomes for example:

```ts
const errors: {
    firstName?: FieldError | undefined;
}
```

I expected the same object shape without a type for the values

### What browsers are you seeing the problem on?

Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
It's a current limitation with V7, fixed in the next major version: https://codesandbox.io/s/error-type-issue-with-react-hook-form-forked-otu0nb?file=/src/App.tsx however currently it's blocked by TS 4.7 update at the V8 branch.

**Reply by Svish** [reply]
Ah, sweet, thanks for quick reply! Then I'll just add a workaround hack for now, and watch for the V8 release 😊👍
